azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/MavenAuthConfiguration.java

```diff
@@ -22,4 +28,39 @@
     private String serverId;
     private String httpProxyHost;
     private String httpProxyPort;
+
+    public void validate() throws InvalidConfigurationException {
```

Comment:
the responsibility of validation should not be assigned to POJO itself. 
i.e. `validate` method should be moved out to some other place.